src/tabs/tab-header-bar.tsx

```diff
@@ -170,6 +170,9 @@ export function TabHeaderBar({
   function renderTabHeader(tab: TabsProps.Tab) {
     const enabledTabsWithCurrentTab = tabs.filter(tab => !tab.disabled || tab.id === activeTabId);
 
+    if (tab.disabled) {
+      tab.href = '';
```

Comment:
Setting it to empty string does not remove the href, it changes it to point to the current page.

You can check how disabled `href` is handled in button component: https://github.com/cloudscape-design/components/blob/main/src/button/internal.tsx